src/commands/context/useDockerContext.ts

```diff
@@ -30,8 +30,20 @@ export async function useDockerContext(actionContext: IActionContext, node?: Con
                 const contextChangedEventPromise = new Promise((resolve) => {
                     const disposable = ext.dockerContextManager.onContextChanged(() => {
                         disposable.dispose();
+                        clearTimeout(timer);
+
                         resolve();
                     });
+
+                    // If 10 seconds pass and it hasn't been updated, force a refresh
+                    // This covers the scenario where the filesystem watchers aren't working (e.g. Codespaces)
+                    const timer = setTimeout(() => {
+                        disposable.dispose();
+                        clearTimeout(timer);
```

Comment:
Do we need the call to `clearTimeout` here (when we are already executing the timeout function)?

Reply:
I suppose not; I was conflating it with `setInterval` which loops by default.